secure/storage/vault/src/lib.rs

```diff
@@ -513,6 +358,230 @@ impl Client {
     }
 }
 
+/// Processes a generic response returned by a vault request. This function simply just checks
+/// that the response was not an error and calls response.into_string() to clear the ureq stream.
+pub fn process_generic_response(resp: Response) -> Result<(), Error> {
+    if resp.ok() {
```

Comment:
according to the doc, this checks

> Whether the response status is: 200 <= status <= 299

do we want to check that the status is 200 exactly instead here?

Reply:
This is a good question for @davidiw, but my preference right now is not to change the logic in this PR. If we want to make a change to logic, I'd suggest doing it in another PR to keep things simple 😄

Reply:
yeah I was mostly commenting on the code I was reading, and this tripped me a bit